addOns/oast/src/main/java/org/zaproxy/addon/oast/services/boast/BoastServer.java

```diff
@@ -61,13 +62,29 @@ public BoastServer(String uriString) throws IOException {
 
         uri = new URI(uriString, true);
         boastMsg = new HttpMessage(uri);
-        boastMsg.getRequestHeader().setHeader(HttpHeader.AUTHORIZATION, generateBoastSecret());
+        secret = generateBoastSecret();
+        boastMsg.getRequestHeader().setHeader(HttpHeader.AUTHORIZATION, "Secret " + secret);
         httpSender.sendAndReceive(boastMsg);
         JSONObject result = JSONObject.fromObject(boastMsg.getResponseBody().toString());
         id = result.getString("id");
         canary = result.getString("canary");
         Stats.incCounter("stats.oast.boast.payloadsGenerated");
     }
+    
+    BoastServer(BoastEntity entity) throws IOException {
+        httpSender =
+                new HttpSender(
+                        Model.getSingleton().getOptionsParam().getConnectionParam(),
+                        true,
+                        // TODO: Replace on next ZAP release with HttpSender.OAST_INITIATOR
+                        ExtensionOast.HTTP_SENDER_OAST_INITIATOR);
+        uri = new URI(entity.getUri(), true);
+        boastMsg = new HttpMessage(uri);
+        secret = entity.getSecret();
+        boastMsg.getRequestHeader().setHeader(HttpHeader.AUTHORIZATION, "Secret " + secret);
```

Comment:
Is there a reason to do an intermediate store here instead of just using getSecret() ?

Reply:
It's being initialized here. Maybe I should use `this.secret` for clarity?